domain/keymanager/state/state.go

```diff
@@ -204,19 +204,23 @@ WHERE user_id = $userId.user_id
 	if err != nil {
 		return nil, fmt.Errorf(
 			"preparing select statement for getting public keys of user %q: %w",
-			id, domain.CoerceError(err),
+			id, err,
 		)
 	}
 
-	userId := userId{id.String()}
+	userId := userId{UserId: id.String()}
 	publicKeys := []publicKey{}
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		return tx.Query(ctx, stmt, userId).GetAll(&publicKeys)
+		err := tx.Query(ctx, stmt, userId).GetAll(&publicKeys)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil
+			}
+			return nil
+		}
```

Comment:
You should really avoid nesting *if* statements like this as you've now introduced a programming error
```suggestion
		if errors.Is(err, sqlair.ErrNoRows) {
			return nil
		} else if err != nil {
			return err
		}
```

Reply:
> like this as you've now introduced a programming error

This informs me that there isn't a test to cover this case.